controllers/argocd/status.go

```diff
@@ -168,6 +168,8 @@ func (r *ReconcileArgoCD) reconcileStatusApplicationSetController(cr *argoproj.A
 		if deploy.Spec.Replicas != nil {
 			if deploy.Status.ReadyReplicas == *deploy.Spec.Replicas {
 				status = "Running"
+			} else if deploy.Status.ReadyReplicas < *deploy.Spec.Replicas {
```

Comment:
@jgwest @Rizwana777 
Personally, I am of the opinion we should overhaul the entire Argo CD status reporting mechanism we currently have. It seems rather unhelpful to me to just have it say "running" or "pending" without offering any more insight into why that is the case. I would like us to be a bit more comprehensive in how we determine the state of a component and also include a reason for why a component is not running successfully 

I think we should ideally be looking at the `DeploymentConditions` in the deployment status to determine what the latest condition is. It also includes the reason if the deployment is failed/unavailable

Reply:
I think image pull errors are 1 of many cases where the deployment could fail
another one could be that the pods fail to come up because they don't meet the security constraints set in place by a cluster (like on openshift)

In such cases I would hope that we can rely on the deploymentconditions to tell us what is wrong and we can relay that directly to the user, as that would be better than us trying to hard code specific checks to determine if the deployment has failed

Reply:
@jaideepr97 you mean instead of checking the state for each pod, we should check for the DeploymentConditions right?

Reply:
Got it @jaideepr97 thank you